internal/shim-sgx/src/enclave.rs

```diff
@@ -0,0 +1,287 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! The Enclave entry and exit
+//!
+//! Provides enclave EENTER entry point and a `syscall` interface.
+use sgx::types::ssa::StateSaveArea;
+
+pub enum Context {}
+
+extern "C" {
+    pub fn syscall(aex: &mut StateSaveArea, ctx: &Context) -> u64;
+}
+
+mod _internal {
+    use crate::event::event;
+    use const_default::ConstDefault;
+    use rcrt1::_dyn_reloc;
+    use syscall::SYS_ENARX_ERESUME;
+    use xsave::XSave;
+
+    const GPR: u64 = 4096 - 184;
+    const RSPO: u64 = GPR + 32;
+    const MISC: u64 = GPR - 16;
+    const SRSP: u64 = MISC - 8;
+
+    const STACK: u64 = 9 * 8;
+    const SHIM: u64 = 10 * 8;
```

Comment:
@npmccallum @lkatalin : maybe you can document these and make suggestions how to create this computationally.

Reply:
Good suggestion - I wasn't able to dig it up from the original PR and can't remember, but I made a debt issue for this: https://github.com/enarx/enarx-keepldr/issues/218